docs/getaddrinfo.md

```diff
@@ -0,0 +1,139 @@
+# Using `getaddrinfo` to Prepare ServerSocket for Binding
+
+We are using the `getaddrinfo` function in the prepareServerSocket method of the ServerSocket object. Before binding the socket to an IP:port combinatio we need to 'prepare' it.
+
+The ServerSocket class contains:
+
+- a Listen object, which contains information relative to the listen directive in the server block in the configuration file.
+- a file descriptor that we get from the socket() fuction
+- a serverSocketAddr variable, which is of type sockaddr_storage. This is an agnostic type that could host IPv4 and IPv6 addresses.
+
+...
+
+The `getaddrinfo` function is an essential tool in network programming, especially when setting up server sockets that need to listen on specific ports and IP addresses. This guide provides an overview of how to use `getaddrinfo` for preparing a `ServerSocket` address and the differences between `getaddrinfo` and `inet_ntop`.
+
+Certainly! Let's continue the exposition in the detailed and informative style you've started:
```

Comment:
Who is this "Certainly!" for ?

Reply:
Yes, I need to review it! It's partially copy/paste. Thanks for reading it!